From 7434790fb162a2f8b81f0e544b0ba450d25a4246 Mon Sep 17 00:00:00 2001
From: Henrik Riomar <henrik.riomar@gmail.com>
Date: Wed, 27 Mar 2024 18:54:53 +0100
Subject: [PATCH] fix DeprecationWarning in test_ecdh

Also update docs to match

Fixes:
 CryptographyDeprecationWarning: Curve argument must be an instance of an EllipticCurve class.
---
 docs/opensc.rst                   | 2 +-
 tests/test_public_key_external.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/docs/opensc.rst b/docs/opensc.rst
index 2c65d4a..b0d7749 100644
--- a/docs/opensc.rst
+++ b/docs/opensc.rst
@@ -175,7 +175,7 @@ Smartcard-HSM can generate a shared key via ECDH key exchange.
 
     # Bob generates a keypair, with their public key encoded for
     # interchange
-    bob_priv = ec.generate_private_key(ec.SECP256R1,
+    bob_priv = ec.generate_private_key(ec.SECP256R1(),
                                         default_backend())
     bob_pub = bob_priv.public_key().public_bytes(
         Encoding.DER,
diff --git a/tests/test_public_key_external.py b/tests/test_public_key_external.py
index 7a378e3..91da0ab 100644
--- a/tests/test_public_key_external.py
+++ b/tests/test_public_key_external.py
@@ -79,7 +79,7 @@ class ExternalPublicKeyTests(TestCase):
 
         # Bob generates a keypair, with their public key encoded for
         # interchange
-        bob_priv = ec.generate_private_key(ec.SECP256R1,
+        bob_priv = ec.generate_private_key(ec.SECP256R1(),
                                            default_backend())
         bob_pub = bob_priv.public_key().public_bytes(
             Encoding.DER,
-- 
2.44.0

